xbmc/weather/WeatherManager.h

```diff
@@ -73,24 +51,37 @@ class CWeatherInfo
   std::string currentWind;
   std::string currentDewPoint;
   std::string currentHumidity;
-  std::string busyString;
-  std::string naIcon;
 };
 
 class CWeatherManager : public CInfoLoader, public ISettingCallback
 {
 public:
-  CWeatherManager(void);
-  ~CWeatherManager(void) override;
-  static bool GetSearchResults(const std::string& strSearch, std::string& strResult);
+  CWeatherManager();
+  ~CWeatherManager() override;
+
+  /*!
+   \brief Retrieve the city name for the specified location from the settings
+   \param iLocation the location index (can be in the range [1..MAXLOCATION])
+   \return the city name (without the accompanying region area code)
+   */
+  std::string GetLocation(int iLocation) const;
 
-  std::string GetLocation(int iLocation);
   const std::string& GetLastUpdateTime() const { return m_info.lastUpdateTime; }
   const ForecastDay& GetForecast(int day) const;
   bool IsFetched();
   void Reset();
 
-  void SetArea(int iLocation);
+  /*!
+   \brief Saves the specified location index to the settings. Call Refresh()
+          afterwards to update weather info for the new location.
+   \param iLocation the new location index (can be in the range [1..MAXLOCATION])
+   */
+  void SetArea(int iLocation) const;
```

Comment:
A `const` setter always makes me wonder. It would be more intuitive to me if `SetArea` and `GetArea` were static member functions, then it's clearer that they modify or use global state.

Reply:
I agree. I will make SetArea and GetArea static.